packages/experimental/Badge/src/BadgeColorTokens.ts

```diff
@@ -1,145 +1,493 @@
+            hcBorderColor: t.colors.brandForeground1,
+          },
+          t,
+        ),
       },
       outline: {
-        color: globalTokens.color.yellow.tint40,
-        iconColor: globalTokens.color.yellow.tint40,
-        borderColor: globalTokens.color.yellow.tint40,
+        ...getOutlineColorProps({ color: globalTokens.color.white, colorDark: globalTokens.color.white }, t),
       },
       tint: {
-        backgroundColor: globalTokens.color.yellow.shade40,
-        color: globalTokens.color.yellow.tint40,
-        iconColor: globalTokens.color.yellow.tint40,
-        borderColor: globalTokens.color.yellow.tint40,
+        ...getTintColorProps(
+          {
+            backgroundColor: globalTokens.color.white,
+            color: globalTokens.color.grey[38],
+            borderColor: globalTokens.color.grey[88],
+            backgroundColorDark: globalTokens.color.grey[29],
+            colorDark: globalTokens.color.grey[68],
+            borderColorDark: globalTokens.color.grey[32],
+          },
+          t,
+        ),
+      },
+      ghost: {
+        ...getGhostColorProps(
+          {
+            color: globalTokens.color.white,
+            colorDark: globalTokens.color.white,
+          },
+          t,
+        ),
       },
     },
   } as BadgeTokens);
+
+export type Colors = {
+  backgroundColor?: ColorValue;
+  color?: ColorValue;
+  iconColor?: ColorValue;
+  borderColor?: ColorValue;
+  backgroundColorDark?: ColorValue;
+  colorDark?: ColorValue;
+  borderColorDark?: ColorValue;
+  hcBackground?: ColorValue;
+  hcColor?: ColorValue;
+  hcBorderColor?: ColorValue;
+};
+/**
+ * A function which returns object of props depending on colors and theme.
+ * @param colors object
+ * @param theme
+ * @returns object of props - backgroundColor, color, iconColor
+ */
+function getFilledColorProps(colors: Colors, theme: Theme) {
+  const { backgroundColor, ...restColors } = colors;
+  const color = restColors.color || theme.colors.neutralForegroundOnBrand;
+  const backgroundColorDark = restColors.backgroundColorDark || backgroundColor;
+  const colorDark = restColors.colorDark || color;
+  const hcBackground = restColors.hcBackground || theme.colors.neutralBackgroundInverted;
+  const hcColor = restColors.hcColor || theme.colors.neutralForegroundInverted;
+  const hcBorderColor = restColors.hcBorderColor || theme.colors.neutralForegroundInverted;
+
+  const themeAppearance = theme.host.appearance;
+  switch (themeAppearance) {
+    case 'light':
+    default:
+      return {
+        backgroundColor: backgroundColor,
+        color: color,
+        iconColor: color,
+      };
+    case 'dark':
+      return {
+        backgroundColor: backgroundColorDark,
+        color: colorDark,
+        iconColor: colorDark,
+      };
+    case 'highContrast':
+      return {
+        backgroundColor: hcBackground,
+        color: hcColor,
+        iconColor: hcColor,
+        borderColor: hcBorderColor,
+      };
```

Comment:
These lines seem to be duplicated between all of the types of badges. I think you could consolidate them by filling in "undefined" for the entries that are empty for a particular badge type and then passing the information into a function which generates this object for you.